plugins/woocommerce/src/Admin/WCAdminHelper.php

```diff
@@ -150,6 +150,16 @@ public static function is_store_page( $url = '' ) {
 		}
 		$normalized_path = self::get_normalized_url_path( $url );
 
+		$exact_matches = array(
+			'?post_type=product'
+		);
+
+		foreach ($exact_matches as $exact_match) {
+			if ( 0 === strpos( $normalized_path, $exact_match ) ) {
```

Comment:
I think this might not be the best way to go around it since the URL may contain more parameters.

For example, I can bypass this logic by using the URL: `/?post_type=product&s=`

ChatGPT suggests using `wp_parse_url` and `wp_parse_str` to obtain the exact query variable and then we can properly compare it.

Reply:
@ilyasfoo Good catch. I did not realize `get_normalized_url_path` does not preserve the param orders. It returns `?s=a&post_type=product` when the URl is `/?post_type=product&s=`

I think str_contains might be good enough. I've updated it to match post_type in [Match post types](https://github.com/woocommerce/woocommerce/pull/50567/commits/9222b460d9e6493289f0e392dbcb40965213b487)

Reply:
Nice, this tests well now! Thanks